### PR TITLE
Improve invite handling

### PR DESCRIPTION
**Pull request by RipleyTom**

### CODE REVIEW

File: Utilities/StrFmt.cpp [hunk 1]
```diff
@@ -214,6 +214,25 @@ void fmt_class_string<std::vector<char8_t>>::format(std::string& out, u64 arg)
...
+template <>
+void fmt_class_string<fmt::buf_to_hexstring>::format(std::string& out, u64 arg)
+{
+	const auto& _arg = get_object(arg);
+	out.reserve(out.size() + (_arg.len * 3));
+	static constexpr char hex[16] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};
+
+	for (usz index = 0; index < _arg.len; index++)
+	{
+		out += hex[_arg.buf[index] >> 4];
```
Comment (elad335) [inline Utilities/StrFmt.cpp:226]: Load it once from the memory please, this is for use of vm memory in case other threads write to it - so it won't print values which were never present in memory and cause confusion.

File: rpcs3/Emu/Cell/Modules/sceNp.cpp [hunk 2]
```diff
@@ -1103,19 +1106,28 @@ error_code sceNpBasicRecvMessageAttachmentLoad(u32 id, vm::ptr<void> buffer, vm:
...
+		return SCE_NP_BASIC_ERROR_INVALID_DATA_ID;
+	}
+
+	const auto opt_msg = nph.get_message_selected(id);
 	if (!opt_msg)
 	{
 		return SCE_NP_BASIC_ERROR_INVALID_DATA_ID;
 	}
 
+	// nph.clear_message_selected(id);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/sceNp.cpp:1120]: Missing comment. Why is this here?

File: rpcs3/Emu/Cell/Modules/sceNp.cpp [hunk 3]
```diff
@@ -1180,11 +1192,17 @@ error_code sceNpBasicRecvMessageCustom(u16 mainType, u32 recvOptions, sys_memory
...
+	att_data->data.id = (mainType == SCE_NP_BASIC_MESSAGE_MAIN_TYPE_INVITE) ? SCE_NP_BASIC_SELECTED_INVITATION_DATA : SCE_NP_BASIC_SELECTED_MESSAGE_DATA;
 	att_data->data.size                        = static_cast<u32>(msg.data.size());
 	att_data->userAction                       = recv_result;
 	att_data->markedAsUsed                     = (recvOptions & SCE_NP_BASIC_RECV_MESSAGE_OPTIONS_PRESERVE) ? 0 : 1;
 
+	extra_nps::print_SceNpBasicExtendedAttachmentData(att_data);
+
+	nph.set_message_selected(att_data->data.id, chosen_msg_id);
+
+	// sysutil_send_system_cmd(CELL_SYSUTIL_NP_INVITATION_SELECTED, 0);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/sceNp.cpp:1204]: Also missing comment

File: rpcs3/rpcs3qt/recvmessage_dialog_frame.cpp [hunk 4]
```diff
@@ -85,6 +85,8 @@ bool recvmessage_dialog_frame::Exec(SceNpBasicMessageMainType type, SceNpBasicMe
 		{ accept_or_deny(SCE_NP_BASIC_MESSAGE_ACTION_ACCEPT); });
     connect(btn_deny, &QAbstractButton::clicked, this, [&accept_or_deny]()
 		{ accept_or_deny(SCE_NP_BASIC_MESSAGE_ACTION_DENY); });
+    connect(btn_cancel, &QAbstractButton::clicked, this, [m_dialog=this->m_dialog]()
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/recvmessage_dialog_frame.cpp:88]: Weird formatting, just capture [this].
Or just connect it to the cancel slot of m_dialog

File: rpcs3/rpcs3qt/recvmessage_dialog_frame.cpp [hunk 5]
```diff
@@ -85,6 +85,8 @@ bool recvmessage_dialog_frame::Exec(SceNpBasicMessageMainType type, SceNpBasicMe
 		{ accept_or_deny(SCE_NP_BASIC_MESSAGE_ACTION_ACCEPT); });
     connect(btn_deny, &QAbstractButton::clicked, this, [&accept_or_deny]()
 		{ accept_or_deny(SCE_NP_BASIC_MESSAGE_ACTION_DENY); });
+    connect(btn_cancel, &QAbstractButton::clicked, this, [this]()
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/recvmessage_dialog_frame.cpp:88]: The indentation of both connects seems wrong 🤨
